### ISSUE TITLE
Задание 3

### ISSUE THREAD

**Issue by AlertRED** [AlertRED/phpTasks#3]

**Comment by mx2s** [comment]
https://github.com/AlertRED/phpTasks/blob/0b14c3343da71a70ce7a9ada5659149470e126ab/routes/api.php#L17
не совсем логично - работу с группами лучше вынести в отдельный контроллер, это ко всем 5 роутам относится

**Comment by mx2s** [comment]
![image](https://user-images.githubusercontent.com/21367904/53882256-e49dea00-4026-11e9-92ca-c2d432ebe411.png)
код плохо читается - старайся следить за этим)

**Comment by mx2s** [comment]
при запросе POST /api/v0/users/group
```
"message": "Class 'App\\Http\\Controllers\\v0\\Validator' not found",
```
идеальный вариант это выносить валидацию в реквесты https://laravel.com/docs/master/validation#form-request-validation

**Comment by mxss** [comment]
https://github.com/AlertRED/phpTasks/blob/0b14c3343da71a70ce7a9ada5659149470e126ab/app/Http/Controllers/v0/UserProfilesController.php#L40
переменной $id не существует в контексте

**Comment by mx2s** [comment]
https://github.com/AlertRED/phpTasks/blob/0b14c3343da71a70ce7a9ada5659149470e126ab/app/Http/Controllers/v0/UserProfilesController.php#L108
переменная $result нигде не используется

https://github.com/AlertRED/phpTasks/blob/0b14c3343da71a70ce7a9ada5659149470e126ab/app/Http/Controllers/v0/UserProfilesController.php#L92
$v то же самое - когда валидацию выполняешь она выкинет exception автоматически - не нужно получать результат и проверять (но на счет того как ты сейчас валидацию делаешь я на 100% не уверен)

**Comment by AlertRED** [comment]
> [phpTasks/routes/api.php](https://github.com/AlertRED/phpTasks/blob/0b14c3343da71a70ce7a9ada5659149470e126ab/routes/api.php#L17)
> 
> Line 17 in [0b14c33](/AlertRED/phpTasks/commit/0b14c3343da71a70ce7a9ada5659149470e126ab)
> 
>  	Route::post('users/group', 'v0\UserProfilesController@postGroup'); 
> 
> не совсем логично - работу с группами лучше вынести в отдельный контроллер, это ко всем 5 роутам относится

Да, я в следующей ветке это исправил почти сразу. Я видел что это не логично, просто сомневался, потому что ожидал, что об этом будет явно сказано в задании.

**Comment by AlertRED** [comment]
> ![image](https://user-images.githubusercontent.com/21367904/53882256-e49dea00-4026-11e9-92ca-c2d432ebe411.png)
> код плохо читается - старайся следить за этим)

Хорошо, я что-то придумаю. Я так и не смог найти автоформатирования кода пыхи в саблайме.

**Comment by AlertRED** [comment]
> при запросе POST /api/v0/users/group
> 
> ```
> "message": "Class 'App\\Http\\Controllers\\v0\\Validator' not found",
> ```
> идеальный вариант это выносить валидацию в реквесты https://laravel.com/docs/master/validation#form-request-validation

Мне не нравилось что к каждому методу нужно создавать файл в котором не существенная проверка (проверка на валидность). Но я все таки смирился)

**Comment by mx2s** [comment]
> просто сомневался, потому что ожидал, что об этом будет явно сказано в задании.

сомневаешься - спрашивай)

**Comment by mx2s** [comment]
> Мне не нравилось что к каждому методу нужно создавать файл в котором не существенная проверка (проверка на валидность)

мне на самом деле тоже, но когда проект будет расти это будет все таки удобнее
